### PR TITLE
Refactor & adding one more parameter to GetLatestVersion

### PR DESCRIPTION
**Pull request by tomahawk28**

### CODE REVIEW

File: appcatalog.go [hunk 1]
```diff
@@ -16,60 +16,21 @@ import (
...
-
-	return "", microerror.Maskf(notFoundError, "no app %#q in index.yaml with given appVersion %#q", app, appVersion)
+	return entry.Urls[0], nil
 }
 
 // GetLatestVersion returns the latest app version for the specified storage URL and app
 // and returns notFoundError when it can't find a specified app.
-func GetLatestVersion(ctx context.Context, storageURL, app string) (string, error) {
-	index, err := getIndex(storageURL)
+func GetLatestVersion(ctx context.Context, storageURL, app, appVersion string) (string, error) {
```
Comment (tomahawk28) [inline appcatalog.go:28]: This is the actual change. I would like to filter the result from `appVersion`.

Comment (rossf7) [reply appcatalog.go:28]: It's a breaking change but we're the only ones using it AFAIK. 👍

File: appcatalog.go [hunk 2]
```diff
@@ -16,60 +16,21 @@ import (
...
-	{
-		entry, ok := index.Entries[app]
-		if !ok {
-			return "", microerror.Maskf(notFoundError, "no app %#q in index.yaml", app)
-		}
-		version = entry[0].Version
-	}
-
-	return version, nil
+	return entry.Version, nil
```
Comment (rossf7) [inline appcatalog.go:33]: Please keep the newline like we had before :)

Comment (tomahawk28) [reply appcatalog.go:33]: Good catch, Thanks for the hint!

**Pull request merged**